source/MRMesh/MRPointsProject.h

```diff
@@ -54,5 +56,51 @@ MRMESH_API void findFewClosestPoints( const Vector3f& pt, const PointCloud& pc,
 /// finds two closest points (first id < second id) in whole point cloud
 [[nodiscard]] MRMESH_API VertPair findTwoClosestPoints( const PointCloud& pc, const ProgressCallback & progress = {} );
 
+/// settings for \ref IPointsProjector::findProjections
+struct FindProjectionOnPointsSettings
+{
+    /// bitset of valid input points
+    const BitSet* valid = nullptr;
+    /// affine transformation for input points
+    const AffineXf3f* xf = nullptr;
+    /// upper limit on the distance in question, if the real distance is larger than the function exits returning upDistLimitSq and no valid point
+    float upDistLimitSq = FLT_MAX;
+    /// low limit on the distance in question, if a point is found within this distance then it is immediately returned without searching for a closer one
+    float loDistLimitSq = 0.f;
+    /// if true, discards a projection candidate with the same index as the target point
+    bool skipSameIndex = false;
+    /// progress callback
+    ProgressCallback cb;
+};
+
+/// abstract class for computing the closest points of point clouds
+class IPointsProjector
```

Comment:
should we unify projector interface for mesh and point cloud (mb other time, but we should think about it)